src/amo/sagas/landing.js

```diff
@@ -10,20 +10,29 @@ import { LANDING_PAGE_ADDON_COUNT } from 'amo/constants';
 import { featured as featuredApi } from 'core/api';
 import { search as searchApi } from 'core/api/search';
 import {
-  LANDING_GET, SEARCH_SORT_POPULAR, SEARCH_SORT_TOP_RATED,
+  LANDING_GET,
+  SEARCH_SORT_POPULAR,
+  SEARCH_SORT_TOP_RATED,
 } from 'core/constants';
 import log from 'core/logger';
 import { createErrorHandler, getState } from 'core/sagas/utils';
 
 
 export function* fetchLandingAddons(
-  { payload: { errorHandlerId, addonType } }
+  { payload: { addonType, category, errorHandlerId } }
 ) {
   const errorHandler = createErrorHandler(errorHandlerId);
   try {
     const state = yield select(getState);
     const { api } = state;
-    const filters = { addonType, page_size: LANDING_PAGE_ADDON_COUNT };
+    const filters = {
+      addonType,
+      page_size: LANDING_PAGE_ADDON_COUNT,
+    };
+
+    if (category !== null) {
```

Comment:
Maybe I'm missing it (the diff is a bit tough to read because of the re-org, but that's nice!), but I don't see a test for this. If the category is false-y at all, really, it shouldn't be set, and we should make sure it's not passed as a filter in that case.

Reply:
I have updated the condition. There are two tests, one with the category specified, the other without. I have added a new test to cover the case where `category` is given with `undefined` as value.